src/codegen/compvars.cpp

```diff
@@ -330,6 +330,57 @@ class UnknownType : public ConcreteCompilerType {
         return new ConcreteCompilerVariable(UNKNOWN, rtn, true);
     }
 
+    CompilerVariable* getiter(IREmitter& emitter, const OpInfo& info, ConcreteCompilerVariable* var) override {
+        static const std::string hasnext_str("__hasnext__");
+        static const std::string iter_str = "__iter__";
+
+        llvm::AllocaInst* rtn_value_memory = emitter.getBuilder()->CreateAlloca(g.llvm_value_type_ptr);
```

Comment:
Rather than doing an alloca+store+load, we should use a PHI node (grep for "PHI").  Also, if we want stack memory, we should place the alloca in the function entry block so that it's a static alloca (becomes part of the frame's stack size) rather than put it right with where it's used, since if this is in a for loop we will alloca additional stack space for every iteration.